syntax = "proto3";
// python -m grpc_tools.protoc -I./ --python_out=pyi_out:. --grpc_python_out=. sysInfo.proto
package sysInfo;

message CpuRequest {
}

message CpuReply {
    float usage = 1;
}

message MemoryRequest {

}

message MemoryReply {
    int32 usage = 1;
    int32 total = 2;
    int32 used = 3;
}

message SysInfoRequest {

}

message CpuInfo {
    string cpuName = 1;
    int32 cpuCore = 2;
    int32 cpuThread = 3;
    float cpuFreq = 4;
}

message MemoryInfo {
    float memoryTotal = 1;
}


message SysInfoReply {
    CpuInfo cpuInfo = 1;
    MemoryInfo memoryInfo = 2;
    message diskInfo {

    }
}


service SysInfoService {
    rpc cpuService (CpuRequest) returns (stream CpuReply) {}
    rpc sysInfoService (SysInfoRequest) returns (SysInfoReply) {}
}